plc4j/api/src/main/java/org/apache/plc4x/java/api/PlcConnectionManager.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.plc4x.java.api;
+
+import org.apache.plc4x.java.api.authentication.PlcAuthentication;
+import org.apache.plc4x.java.api.exceptions.PlcConnectionException;
+
+public interface PlcConnectionManager {
+
+    /**
+     * Connects to a PLC using the given plc connection string.
+     *
+     * @param url plc connection string.
+     * @return PlcConnection object.
+     * @throws PlcConnectionException an exception if the connection attempt failed.
+     */
+    PlcConnection getConnection(String url) throws PlcConnectionException;
+
+    /**
+     * Connects to a PLC using the given plc connection string using given authentication credentials.
+     *
+     * @param url            plc connection string.
+     * @param authentication authentication credentials.
+     * @return PlcConnection object.
+     * @throws PlcConnectionException an exception if the connection attempt failed.
+     */
+    PlcConnection getConnection(String url, PlcAuthentication authentication) throws PlcConnectionException;
```

Comment:
@chrisdutz we should some day think about using the option pattern from Golang here to (the ones with the varargs options). This way we could combine the getConnection into one method.

Reply:
Well that is incompatible with the try-with-resources pattern ...

Reply:
At least I think so

Reply:
Don't think that should be a Problem but maybe I don't see it :D